setup.py

```diff
@@ -155,5 +155,4 @@ def get_scm_config():
         ],
     },
     use_scm_version=get_scm_config,
-    setup_requires=['setuptools_scm'],
```

Comment:
Ah, I guess we'll now have to install those manually when using plain `setup.py` based installation? We can either change the targets in the Makefile to invoke something we now need or update the CI scripts to install the necessary deps before invoking the Makefile target in the (for the infer CI job, other jobs install via `pip install .`)

Reply:
The pip equivalent to `python setup.py develop` is a bit of a pain -- something like `pip install <write-build-requirements-by-hand-here>` followed by `pip install --no-build-isolation -e .`

I'm not sure it actually hurts to duplicate `setuptools_scm` in `setup_requires`. Maybe it would be fine to leave it in?

Reply:
So it looks like we just need to revert the changes in [setup.py](https://github.com/hpyproject/hpy/pull/287/files#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7) and we can merge this. @domdfcoding, would you like to do that? Otherwise we can also take this over if you want.